### PR TITLE
fix: update getPath to use WHATWG URL API

### PR DESCRIPTION
**Pull request by mschile**
### Additional details

* Updated the `getPath` method to use the [WHATWG URL API](https://nodejs.org/api/url.html#the-whatwg-url-api).

When processing requests, we set the [url](https://github.com/cypress-io/cypress/blob/02d04934725afd4a5b8067c1dd8ec1415e99ed3a/packages/server/lib/server-base.ts#L114) using the [legacy url API](https://nodejs.org/api/url.html#legacy-url-api). Since this API is not WHATWG URL API compliant with respect to [percent encoding](https://nodejs.org/api/url.html#percent-encoding-in-urls), we receive a path that may be different from what CDP sends causing prerequest correlation timeouts of two seconds.

The following requests were observed in the wild:

With the encoded `|` as `%7C`:
`https://fonts.googleapis.com/css?family=Roboto+Mono%7COpen+Sans:700&display=swap`

Already decoded `|`:
`https://fonts.googleapis.com/css?family=Roboto+Mono|Open+Sans:700&display=swap`

### PR Tasks

- [x] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?

### CODE REVIEW

**Review by ryanthemanuel — APPROVED** [review]

File: packages/network/lib/uri.ts [hunk 1]
```diff
@@ -87,7 +87,9 @@ export function addDefaultPort (urlToCheck: any) {
 }
 
 export function getPath (urlToCheck: string) {
-  return url.parse(urlToCheck).path
+  const url = new URL(urlToCheck)
```
Comment (emilyrohrbough) [inline packages/network/lib/uri.ts]: There are several places in this file using `url.parse` still. What not update those?

Comment (mschile) [reply packages/network/lib/uri.ts]: Yeah, we should go through the entire codebase and update to the newer URL API. I wanted to address the performance issue in a timely manner with minimum risk. I'll log an issue to update to the new API.

Comment (emilyrohrbough) [reply packages/network/lib/uri.ts]: Gottchya---I assumed we'd be at risk with parsing these URLs differently if they return different results?

Comment (mschile) [reply packages/network/lib/uri.ts]: `getPath` is only used in one location when resetting the [url](https://github.com/cypress-io/cypress/blob/02d04934725afd4a5b8067c1dd8ec1415e99ed3a/packages/server/lib/server-base.ts#L114). All the other functions in this file are used elsewhere and don't make direct comparisons to the url.

**Review by emilyrohrbough — APPROVED** [review]

File: packages/network/lib/uri.ts (hunk 1 above)
Comment (mschile) [reply packages/network/lib/uri.ts]: logged https://github.com/cypress-io/cypress/issues/28369

**Pull request merged**